app/src/main/java/com/hedvig/app/feature/insurance/ui/detail/coverage/PerilBottomSheet.kt

```diff
@@ -2,68 +2,204 @@ package com.hedvig.app.feature.insurance.ui.detail.coverage
+import android.widget.ImageView
 import androidx.core.os.bundleOf
+import com.google.android.material.bottomsheet.BottomSheetBehavior
+import com.google.android.material.bottomsheet.BottomSheetBehavior.BottomSheetCallback
+import com.google.android.material.bottomsheet.BottomSheetBehavior.STATE_DRAGGING
+import com.google.android.material.bottomsheet.BottomSheetDialog
 import com.google.android.material.bottomsheet.BottomSheetDialogFragment
 import com.hedvig.android.owldroid.fragment.PerilFragment
 import com.hedvig.app.BuildConfig
 import com.hedvig.app.R
 import com.hedvig.app.databinding.PerilBottomSheetBinding
+import com.hedvig.app.util.extensions.colorAttr
+import com.hedvig.app.util.extensions.dp
 import com.hedvig.app.util.extensions.isDarkThemeActive
+import com.hedvig.app.util.extensions.view.setHapticClickListener
 import com.hedvig.app.util.extensions.viewBinding
-import com.hedvig.app.util.svg.buildRequestBuilder
+import com.hedvig.app.util.safeLet
 import e
+import java.util.ArrayList
 
 class PerilBottomSheet : BottomSheetDialogFragment() {
     private val binding by viewBinding(PerilBottomSheetBinding::bind)
     override fun onCreateView(
         inflater: LayoutInflater,
         container: ViewGroup?,
         savedInstanceState: Bundle?
-    ): View =
-        inflater.inflate(R.layout.peril_bottom_sheet, container, false)
+    ): View = inflater.inflate(R.layout.peril_bottom_sheet, container, false)
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        val defaultStatusBarColor = dialog?.window?.statusBarColor
+        val defaultSystemUiVisibility = dialog?.window?.decorView?.systemUiVisibility
+        val bottomSheetDialog = super.onCreateDialog(savedInstanceState) as BottomSheetDialog
+        val title = requireArguments().getString(TITLE)
+        val description = requireArguments().getString(DESCRIPTION)
+        val iconUrl = requireArguments().getString(ICON_URL)
+        val exception = requireArguments().getStringArrayList(EXCEPTIONS)
+        val covered = requireArguments().getStringArrayList(COVERED)
+        val info = requireArguments().getString(INFO)
+
         binding.apply {
-            val titleText = requireArguments().getString(TITLE)
-            val bodyText = requireArguments().getString(BODY)
-            val iconUrl = requireArguments().getString(ICON_URL)
+            close.alpha = 0f
+            (dialog as? BottomSheetDialog)?.behavior?.setPeekHeight(380.dp, true)
+
+            dialog?.setOnShowListener { dialogInterface ->
+                val containerLayout =
+                    (dialogInterface as BottomSheetDialog).findViewById<FrameLayout>(com.google.android.material.R.id.container)
+                val shadow =
+                    bottomSheetDialog.layoutInflater.inflate(R.layout.bottom_sheet_shadow, null)
+                val chevron = shadow.findViewById<ImageView>(R.id.chevron)
+                chevron.setHapticClickListener { expandSheet() }
```

Comment:
Do we need to do these things? Would it not be possible to add this in XML, in `R.layout.peril_bottom_sheet`?

Reply:
Probably, but i thought this was a better solution now its always in the bottom of the bottomsheet parent

Reply:
I worry that this solution might be a bit brittle, given that it more or less relies on implementation details.

I'd much prefer that we add this to the XML, unless there's something truly preventing us from doing that.